lib/tasks/conditions/process_conditions.rake

```diff
@@ -0,0 +1,30 @@
+require 'active_support/logger'
+
+desc 'process conditions'
+task :process_conditions => [:environment] do
+
+  LOG = 'log/conditions'
+
+  ActivityLogger.open(LOG, 'SHF_TASK', 'Conditions') do |log|
+
+    begin
+      class_name = nil
+
+      Condition.order(:class_name).each do |condition|
+
+        unless condition.class_name == class_name
+          klass = condition.class_name.constantize
+          class_name = condition.class_name
+        end
+
+        log.record('info', "#{class_name}: #{condition.name} ...")
+
+        klass.condition_response(condition, log)
```

Comment:
`klass` can be undefined here (possible, but not probable I know).  (because of the `unless....` conditional + it's not initialized_

Need a simple statement to handle this (e.g. initialize `klass` to something (Condition class?)

Reply:
Good point.  I added logic to handle: 1) class_name = `nil` (for the situation you mention (although we validate the presence of `class_name` in the model)), and, 2) entering an incorrect class_name in the record.

For the first the log will contain:
```
[SHF_TASK] [Conditions] [info] Started at 2018-12-07 12:52:27 UTC
[SHF_TASK] [Conditions] [info] : membership_will_expire ...
[SHF_TASK] [Conditions] [error] Exception: #<RuntimeError: klass not assigned in rake task shf:process_conditions>
[SHF_TASK] [Conditions] [info] Finished at 2018-12-07 12:52:27 UTC.
[SHF_TASK] [Conditions] [info] Duration: 0.04 seconds.
```

And for the 2nd:
```
[SHF_TASK] [Conditions] [info] Started at 2018-12-07 12:57:14 UTC
[SHF_TASK] [Conditions] [error] Exception: #<NameError: uninitialized constant NoneSuch>
[SHF_TASK] [Conditions] [info] Finished at 2018-12-07 12:57:14 UTC.
[SHF_TASK] [Conditions] [info] Duration: 0.04 seconds.
```